courses/rust/projects/project-2/README.md

```diff
@@ -187,18 +187,18 @@ is crucial to Rust projects: decide on an error handling strategy.
 <!-- TODO outline strategies? -->
 
 Rust's error handling is powerful, but involves a lot of boilerplate to use
-correctly. For this project the [`failure`] crate will provide the tools to
+correctly. For this project the [`thiserror`] crate will provide the tools to
 easily handle errors of all kinds.
 
-[`failure`]: https://docs.rs/failure/0.1.5/failure/
+[`thiserror`]: https://docs.rs/thiserror/1.0.18/thiserror/
 
 The [failure guide][fg] describes [several] error handling patterns.
```

Comment:
This failure guide might be removed or replaced with other materials based on `thiserror` crate.

Reply:
Well, I asked brain anderson about this, but he didn't reply, this part shows patterns for using error handling, I think replace it maybe not good.
Would you mind give some suggestion here?

Reply:
Maybe I can add strikethrough on contents about `thiserror` @ziyi-yan